src/commands.c

```diff
@@ -46,29 +46,35 @@ void imv_command_alias(struct imv_commands *cmds, const char *command, const cha
   list_append(cmds->command_list, cmd);
 }
 
-int imv_command_exec(struct imv_commands *cmds, const char *command, void *data)
+int imv_command_exec(struct imv_commands *cmds, struct list *commands, void *data)
```

Comment:
I'd prefer to leave `imv_command_exec` more or less as-is, and then have a `imv_command_exec_list` function that does the iteration. That way command execution isn't tied in with handling a list of commands.

Reply:
Good idea, done :)